cloud/cluster/memory/cluster.go

```diff
@@ -0,0 +1,138 @@
+package memory
```

Comment:
The polling cluster ends up being almost the inverse of this, where it polls and then creates updates, but because it's polling, it has its own state of the current members, as well. So I think it may actually make sense in production to skip with this code altogether, to not have a second pool of current nodes. So, yes, I think this implementation is not necessary in production, and thus belongs in a separate package.